lib/publiccloud/ecr.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Amazon Elastic Container Registry (ECR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
+
+package publiccloud::ecr;
+use Mojo::Base 'publiccloud::aws';
+use testapi;
+
+sub push_container_img {
```

Comment:
Maybe you should add a stub in `publiccloud::provider` as well like: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L92

Reply:
If this will be a method, which should be implemented by other container implementations...